app/src/main/java/com/x8bit/bitwarden/ui/platform/base/util/StringResExtensions.kt

```diff
@@ -0,0 +1,142 @@
+
+/**
+ * Creates an [AnnotatedString] from a string resource allowing for optional arguments
+ * to be applied.
+ * @param args Optional arguments to be applied to the string resource, must already be a
+ * [String]
+ * @param style Style to apply to the entire string
+ * @param emphasisHighlightStyle Style to apply to part of the resource that has been annotated
+ * with the "emphasis" annotation.
+ * @param linkHighlightStyle Style to apply to part of the resource that has been annotated with
+ * the "link" annotation.
+ * @param onAnnotationClick Callback to invoke when a link annotation is clicked. Will pass back
+ * the value of the annotation as a string to allow for delineation if there are multiple callbacks
+ * to be applied.
+ *
+ * In order for the styles to be applied the resource must contain custom annotations for example:
+ *
+ * If a word or phrase is to have the [emphasisHighlightStyle] applied then it must be annotated
+ * with the custom XML tag: <annotation emphasis="anything"> where the value does not matter.
+ * <string foo>Foo <annotation emphasis="anything">bar</annotation> baz</string>
+ *
+ * If a word or phrase is to have the [linkHighlightStyle] applied then it must be annotated
+ * with the custom XML tag: <annotation link="callBackKey"> where the value will be passed back
+ * in [onAnnotationClick] and used to delineate which annotation was clicked.
+ * <string foo>Foo <annotation link="onBarClick">bar</annotation> baz</string>
+ *
+ * If the <string> contains a format argument (%1$s) then that argument should be wrapped in the
+ * following custom XML tag: <annotation arg="0"> where the value is the index of the argument,
+ * starting at 0.
+ */
+@Suppress("LongMethod")
+@Composable
+fun @receiver:StringRes Int.toAnnotatedString(
+    vararg args: String,
+    style: SpanStyle = bitwardenDefaultSpanStyle,
+    emphasisHighlightStyle: SpanStyle = bitwardenBoldSpanStyle,
+    linkHighlightStyle: SpanStyle = bitwardenClickableTextSpanStyle,
+    onAnnotationClick: ((annotationKey: String) -> Unit)? = null,
+): AnnotatedString {
+    val resources = LocalContext.current.resources
+    val spannableBuilder = try {
+        SpannableStringBuilder(resources.getText(this) as SpannedString)
+    } catch (e: ClassCastException) {
+        return stringResource(id = this).toAnnotatedString()
+    }
+    spannableBuilder.applyArgAnnotations(args = args)
+    val annotatedStringBuilder = AnnotatedString.Builder()
+    annotatedStringBuilder.append(spannableBuilder)
+    annotatedStringBuilder.addStyle(
+        style = style,
+        start = 0,
+        end = spannableBuilder.length,
+    )
+    spannableBuilder.getSpans<Annotation>(0, spannableBuilder.length)
+        .forEach { annotation ->
+            val start = spannableBuilder.getSpanStart(annotation)
+            val end = spannableBuilder.getSpanEnd(annotation)
+            when (annotation.key) {
+                "emphasis" -> {
```

Comment:
🔨 should this be a `enum`?

Reply:
I had them, but it basically served no other purpose than doing something like:
```
when (val enum = EnumType.valueOf(annotation.key.uppercase())) {
     OPTION1 -> { ... }
```
and if `annotation.key` was not an enum value it would crash. I guess maybe that would be a good thing as that type of error is something that should be caught while testing so would be useful to the developer to be like hey you used an annotation we weren't ready for...and now I am thinking we should add as an enum :)